crypto/objects/obj_dat.c

```diff
@@ -416,26 +432,31 @@ int OBJ_obj2txt(char *buf, int buf_len, const ASN1_OBJECT *a, int no_name)
     unsigned long l;
     const unsigned char *p;
     char tbuf[DECIMAL_SIZE(i) + DECIMAL_SIZE(l) + 2];
+    const char *s;
 
     /* Ensure that, at every state, |buf| is NUL-terminated. */
-    if (buf && buf_len > 0)
+    if (buf != NULL && buf_len > 0)
         buf[0] = '\0';
 
-    if ((a == NULL) || (a->data == NULL))
+    if (a == NULL || a->data == NULL)
+        return 0;
+
+    if (!ossl_obj_read_lock(1))
```

Comment:
Again why do we need to hold a lock for the whole operation rather than let the lower level functions acquire it as required?